upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
```

Comment:
It's generally more efficient to reuse an XMLHttpRequest object or use a different modern approach like Fetch API with stream uploads. Creating a new XMLHttpRequest for every chunk adds unnecessary overhead.